core/src/test/java/org/infinispan/api/AsyncAPITest.java

```diff
@@ -339,29 +349,52 @@ private void verifyEviction(final String key, final String expectedValue, final
          throw new IllegalStateException("markStartTime() must be called before verifyEviction(..)");
       }
 
-      final long pollInterval = 50;
       try {
-         assertTrue(expectedValue.equals(c.get(key)) || TestingUtil.moreThanDurationElapsed(startTime, expectedLifetime));
-         eventually(new Condition() {
-            @Override
-            public boolean isSatisfied() {
-               if (touchKey) {
-                  return !c.containsKey(key);        //this check DOES read the key so it resets the idle time
-               } else {
-                  //this check DOES NOT read the key so it does not interfere with idle time
-                  InternalCacheEntry entry = c.getAdvancedCache().getDataContainer().peek(key);
-                  return entry == null || entry.isExpired(System.currentTimeMillis());
-               }
+         long expectedEndTime = startTime + expectedLifetime;
+         Condition condition = () -> {
+            if (touchKey) {
+               return !c.containsKey(key);        //this check DOES read the key so it resets the idle time
+            } else {
+               //this check DOES NOT read the key so it does not interfere with idle time
+               InternalCacheEntry entry = c.getAdvancedCache().getDataContainer().peek(key);
+               return entry == null || entry.isExpired(timeService.wallClockTime());
             }
-         }, 3 * expectedLifetime, pollInterval, TimeUnit.MILLISECONDS);
-         long waitTime = Util.currentMillisFromNanotime() - startTime;
+         };
+         assertTrue(expectedValue.equals(c.get(key)) || timeService.isTimeExpired(expectedEndTime));
+         while (!condition.isSatisfied() || !timeService.isTimeExpired(expectedEndTime)) {
+            timeService.advance(Math.max(1, expectedLifetime / 10));
```

Comment:
Do you still need to repeat the loop 10 times? It should be enough to advance once to `expectedLifetime * 9 / 10`